File: lib/domain/usecase/search_in_book_use_case.dart
import '../../core/model/book_search_result.dart';
import '../repository/book_repository.dart';

class SearchInBookUseCase {
  final BookRepository _bookRepository;

  SearchInBookUseCase(this._bookRepository);

  Future<List<BookSearchResult>> call(String bookId, String query) =>
      _bookRepository.searchInBook(bookId, query);
}

File: domain/src/main/kotlin/com/readler/domain/usecase/SearchInBookUseCase.kt
package com.readler.domain.usecase

import com.readler.core.model.BookSearchResult
import com.readler.domain.repository.BookRepository

class SearchInBookUseCase(
    private val bookRepository: BookRepository
) {
    suspend operator fun invoke(bookId: String, query: String): List<BookSearchResult> {
        return bookRepository.searchInBook(bookId, query)
    }
}
